src/client/dfuse/pil4dfs/hook.c

```diff
@@ -306,28 +306,17 @@ determine_lib_path(void)
 	}
 
 	/* with version in name */
-	rc = asprintf(&path_libpthread, "%s/libpthread-%s.so", lib_dir_str, libc_version_str);
-	if (rc < 0) {
-		DS_ERROR(ENOMEM, "Failed to allocate memory for path_libpthread");
-		goto err_1;
-	}
-	if (rc >= PATH_MAX) {
-		free(path_libpthread);
-		path_libpthread = NULL;
+	D_ASPRINTF(path_libpthread, "%s/libpthread-%s.so", lib_dir_str, libc_version_str);
+	if (path_libpthread == NULL)
+		goto err;
+	if (strnlen(path_libpthread, PATH_MAX) >= PATH_MAX) {
+		D_FREE(path_libpthread);
 		DS_ERROR(ENAMETOOLONG, "path_libpthread is too long");
-		goto err_1;
-	}
-	rc = asprintf(&path_libdl, "%s/libdl-%s.so", lib_dir_str, libc_version_str);
-	if (rc < 0) {
-		DS_ERROR(ENOMEM, "Failed to allocate memory for path_libdl");
-		goto err_1;
-	}
-	if (rc >= PATH_MAX) {
-		free(path_libdl);
-		path_libdl = NULL;
-		DS_ERROR(ENAMETOOLONG, "path_libdl is too long");
-		goto err_1;
+		goto err;
 	}
+	D_ASPRINTF(path_libdl, "%s/libdl-%s.so", lib_dir_str, libc_version_str);
+	if (path_libdl == NULL)
+		goto err;
```

Comment:
don't you need another strnlen PATH_MAX check here?

Reply:
This path is shorter than the above one, "%s/libpthread-%s.so", whose length was checked already.